File: java/Board.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author i14032
 */
public class Board {

    private int numBH;
    private int numWG;
    private int numAs;
    public Dadu dice;

    private Planet[][] arrPl = new Planet[10][10];
    private SpaceWarpGate[] arrWG;
    private BlackHole[] arrBH;
    private Astronot[] arrAs;

    public Board(int maxDadu, int wg, int bh, int as) {
        this.numAs = as;
        this.numBH = bh;
        this.numWG = wg;
        arrWG = new SpaceWarpGate[numWG];
        arrBH = new BlackHole[numBH];
        arrAs = new Astronot[numAs];
        dice = new Dadu(maxDadu);
        int locate = 1;
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                arrPl[i][j] = new Planet(i, j, locate);
                //locate++;
                arrPl[i][j].setLocatePl(i, j);
            }
        }
    }

    public void addBlackHole(int idx, BlackHole b) {
        arrBH[idx] = b;
    }

    public void addWarper(int idx, SpaceWarpGate w) {
        arrWG[idx] = w;
    }

    public void addAstronot(int idx, Astronot a) {
        arrAs[idx] = a;
    }

    public int getNumBH() {
        return arrBH.length;
    }

    public int getNumAs() {
        return arrAs.length;
    }

    public int getNumWG() {
        return arrWG.length;
    }

    public Astronot getAsAt(int idx) {
        return arrAs[idx];
    }

    public BlackHole getBH(int idx) {
        return arrBH[idx];
    }

    public SpaceWarpGate getWG(int idx) {
        return arrWG[idx];
    }

    public Planet getPlanet(int idxy, int idxx) {
        return arrPl[idxy][idxx];
    }

}

File: php/Board.php
<?php
	
	# Tugas Besar ADBO (Interstellar) - Space Snake&Ladder
	# Dosen Pembimbing : Pascal Alfadian
	
	# @author Ivan TW - 2014730026 || Riki Rusli (Ketua) - 2014730032 || Daniel Ferdinan - 2014730038 
	# Kelas yang merepresentasikan game board pada permainan Interstellar.
	
	class Board{
		private $numBH;
		private $numWG;
		private $numAs;
		public $dice;
		
		private $arrPl;
		private $arrWG;
		private $arrBH;
		private $arrAs;
		
		# Konstruktor untuk men-generate board permainan secara keseluruhan
		# @param maxDadu maksimum banyak mata dadu yang diinginkan dalam permainan (umumnya 6)
		# @param wg banyaknya warp-gate (tangga) dalam board Interstellar
		# @param bh banyaknya blackhole dalam (ular) board Interstellar
		# @param as banyaknya astronot (pemain) dalam permainan Interstellar
		public function __construct($maxDadu,$wg,$bh,$as){
			$this->dice=new Dadu($maxDadu);
			$this->numBH=$bh;
			$this->numWG=$wg;
			$this->numAs=$as;
			$arrPl=array();
			$arrWG=array();
			$arrBH=array();
			$arrAs=array();
			$locate=1;
			
			for($i=0;$i<10;$i++){
				for($j=0;$j<10;$j++){
					$this->arrPl[$i][$j]= new Planet($i,$j,$locate);
					$this->arrPl[$i][$j]->set_Locate_Pl($i,$j);
				}
			}
		}
		
		#Method yang digunakan untuk mengocok dadu (memanggil method di kelas Dadu)
		public function roll_The_Dice(){
			return $this->dice->roll();
		}
		
		# Method untuk menambahkan blackhole (ular) pada board Interstellar
		# @param idx masukan indeks kotak yang akan ditempati blackhole pada board Interstellar
		# @param b masukan objek blackhole
		public function add_Black_Hole($idx,$hole){
			$this->arrBH[$idx]=$hole;
		}
		
		# Method untuk menambahkan warp-gate (tangga) pada board Interstellar
		# @param idx masukan indeks kotak yang akan ditempati warp-gate pada board Interstellar
		# @param w masukan objek warp-gate
		public function add_Warper($idx,$warp){
			$this->arrWG[$idx]=$warp;
		}
		
		# Method untuk menambahkan astronot (pemain) pada board Interstellar
		# @param idx masukan indeks kotak yang akan ditempati astronot (pemain) pada board Interstellar
		# @param a masukan objek astronot (pemain)
		public function add_Astronot($idx,$astronot){
			$this->arrAs[$idx]=$astronot;
		}
		
		# Getter untuk banyak blackhole (ular) pada board permainan Interstellar
		# @return banyak blackhole pada board permainan Interstellar - int
		public function get_Num_BH(){
			return count($arrBH);
		}
		
		# Getter untuk banyak warp-gate (tangga) pada board permainan Interstellar
		# @return banyak warp-gate pada board permainan Interstellar - int
		public function get_Num_WG(){
			return count($arrWG);
		}
		
		# Getter Astronot pada index spesifik dalam board permainan Interstellar
		# @param idx masukan indeks astronot - int
		# @return objek astronot pada indeks spesifik sesuai parameter - Astronot
		public function get_As_At($idx){
			return $this->arrAs[$idx];
		}
		
		# Getter BlackHole pada index spesifik dalam board permainan Interstellar
		# @param idx masukan indeks BlackHole - int
		# @return objek BlackHole pada indeks spesifik sesuai parameter - BlackHole
		public function get_BH($idx){
			return $this->arrBH[$idx];
		}
		
		# Getter warp-gate pada index spesifik dalam board permainan Interstellar
		# @param idx masukan indeks warp-gate - int
		# @return objek warp-gate pada indeks spesifik sesuai parameter - SpaceWarpGate
		public function get_WG($idx){
			return $this->arrWG[$idx];
		}
		
		# Getter planet (kotak) pada index spesifik dalam board permainan Interstellar
		# @param idxy masukan posisi kotak X sumbu kartesian - int
		# @param idxx masukan posisi kotak Y sumbu kartesian - int
		# @return objek planet pada indeks spesifik sesuai parameter - Planet
		public function get_Planet($idxY,$idxX){
			return $this->arrPl[$idxY][$idxX];
		}
	}
?>
